src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -57,15 +58,32 @@ public void storeMessage(Message message) {
     List<Message> messages = new ArrayList<>();
 
     Query query =
-        new Query("Message")
-            .setFilter(new Query.FilterPredicate("user", FilterOperator.EQUAL, user))
+            new Query("Message")
+                    .setFilter(new Query.FilterPredicate("user", FilterOperator.EQUAL, user))
+                    .addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    return retrievedMessages(results, user, Boolean.TRUE);
+  }
+
+  public List<Message> getAllMessages(){
+    List<Message> messages = new ArrayList<>();
+
+    Query query = new Query("Message")
             .addSort("timestamp", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
+    return retrievedMessages(results, "", Boolean.FALSE);
+  }
+
+  public List<Message> retrievedMessages(PreparedQuery results, String user, Boolean userGiven) {
```

Comment:
Method names tend to be "verbs" (e.g. `getMessages`), so maybe this could be `retrieveMessages`?  Or just also call it `getMessages` since it's conceptually the same as `getMessages`, just with different arguments (and Java allows you to overload method names)?